src/lib/utils/linking.native.js

```diff
@@ -0,0 +1,76 @@
+import branch from 'react-native-branch'
+import { extractQueryParams } from '../share'
+import restart from './restart'
+import extractPathname from './extractPathname'
+
+class LinkingNative {
+  constructor() {
+    this.subscribe()
+  }
+
+  _unsubscribe = null
+
+  _lastClick = null
+
+  _isFirstRun = true
+
+  params = {
+    web3: '',
+    paymentCode: '',
+    code: '',
+    reason: '',
+    event: '',
+  }
+
+  navigationCallbacks = []
+
+  pathname = ''
+
+  subscribe = navigationCallback => {
+    if (navigationCallback) {
+      this.navigationCallbacks.push(navigationCallback)
+    }
+    this._unsubscribe = branch.subscribe(this._listener)
```

Comment:
as i've understand you have to subsribe to the branch just once, then your listener will proxy events to all listeners has been added by the moment of event happened

so in current implementation each event will be duplicated N times where N is the count of .subscribe() calls
and you could unsubsribe just latest one subscription

from my point of view, to implement it correctly you need just check is this._unsubscribe non-null/empty before subscribe to the branch:

```javascript
if (!this._unsubscribe) {
  this._unsubscribe = branch.subscribe(this._listener)
}
```

Reply:
Could you please explain why you think that issue is resolved or doesn't require fixing (according to my comments above) ?

Reply:
The problem does not occur, I do not identify the problem testing the subscribe method, has the same performance with both solutions.

Reply:
> The problem does not occur, I do not identify the problem testing the subscribe method, has the same performance with both solutions.

pls test those cases:
1. add a few subscriptions. emit event. each subscription's callback should be called once
2. add a few subscriptions. unsubscribe. emit event. each subscription's callback shouldn't be called